### DISCUSSION TITLE
Questions about gatsby blocks of code

### DISCUSSION THREAD

**Discussion by liuweiyibai** [gatsbyjs/gatsby#30071] [Help]
Hello, what can I do to have the code block display the code type, just like the code block of the gatsby official document, display the file type and file name.

like this:
<img width="868" alt="WeChat8436dd96ddd6c159c6e56434c9717c5a" src="https://user-images.githubusercontent.com/21965057/110242329-8b0f8380-7f90-11eb-96b6-bd917d8a4c96.png">

**Comment by SabbeRubbish** [comment]
Hi liuweiyibai,

I too had wondered how this could be done.
According to multiple sources, the code block's "info string" (cfr. [this article](https://thesynack.com/posts/codeblock-labels-markdown/) by the SynAck) is not standardized. 
Also, the way things are rendered are quite different between for example Github and Gatsby docs.

Github markdown doesn't support this.
I tried your code block source code (see it on Github [here](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/tutorial/part-one/index.md)) 
~~~
```jsx:title=src/pages/index.js
import React from "react"

export default function Home() {
  return <div style={{ color: `purple`, fontSize: `72px` }}>Hello Gatsby!</div>
}
```
~~~
and in Github it just renders to:
![image](https://user-images.githubusercontent.com/5593135/110652421-a7900380-81bc-11eb-995f-11fec44d2737.png)

However, if you look at code blocks in the Gatsby docs, the styling is quite different, already using the info string:
Github ignores the ``:title=`` attribute. But the Gatsby docs loves it:

![image](https://user-images.githubusercontent.com/5593135/110652601-cf7f6700-81bc-11eb-9818-b2dfa1324d97.png)

So now you know the syntax:
~~~
```<filetype>:title=<title>
~~~

Hope this helps!

**Reply by liuweiyibai** [reply]
Hello, I found a way to add file type tags

```js
const fs = require('fs');

const lables = [
  { lang: 'javascript', tag: 'js', bg: 'rgb(247, 223, 30)' },
  { lang: 'js', tag: 'js', bg: 'rgb(247, 223, 30)' },
  { lang: 'typescript', tag: 'ts', bg: 'rgb(41, 78, 128)', color: '#fff' },
  { lang: 'ts', tag: 'ts', bg: 'rgb(41, 78, 128)', color: '#fff' },
  { lang: 'css', tag: 'css', bg: '#ff9800' },
  { lang: 'scss', tag: 'scss', bg: '#ff9800' },
  { lang: 'jsx', tag: 'jsx', bg: '#61dafb' },
  { lang: 'html', tag: 'html', bg: 'rgb(0, 90, 156)',color:"#fff" },
  {
    lang: 'ignore',
    tag: 'ignore',
    bg: 'rgb(255,55,0)',
    color: '#fff',
  },
  {
    lang: 'nginx',
    tag: 'nginx',
    bg: 'rgb(0,161,0)',
    color: '#fff',
  },
  {
    lang: 'ruby',
    tag: 'ruby',
    bg: '#CC342D',
    color: '#fff',
  },
  { lang: 'bash', tag: 'sh', bg: '' },
  { lang: 'text', tag: 'text', bg: '#fff' },
  { lang: 'shell', tag: 'sh', bg: '' },
  { lang: 'json', tag: 'json', bg: '#8bc34a' },
  { lang: 'yaml', tag: 'yaml', bg: '#8bc34a' },
  { lang: 'diff', tag: 'diff', bg: '#e6ffed' },
  { lang: 'markdown', tag: 'md', bg: 'white' },
  { lang: 'graphql', tag: 'GraphQL', bg: '#e10098', color: '#fff' },
  { lang: 'docker', tag: 'docker', bg: '#2697ed', color: '#fff' },
  { lang: 'go', tag: 'go', bg: '#2697ed', color: '#fff' },
  { lang: 'env', tag: 'env', bg: '#2697ed', color: '#fff' },
  { lang: 'env', tag: 'tag', bg: '#2697ed', color: '#fff' },
];

const lablesStyles = lables
  .map(
    ({ lang, tag, bg = 'white', color = 'black' }) =>
      `.gatsby-highlight[data-language="${lang}"] pre:before {
      content: '${tag}';
      ${bg && `background: ${bg};`}
      ${color && `color: ${color};`}
    }`
  )
  .join(`\n`);

fs.writeFileSync('./label.css', lablesStyles);

```

I found a gatsby plug-in to add a title to the article

```bash
yarn add gatsby-remark-code-titles
```
